service/src/main/java/bio/terra/workspace/app/controller/ControlledAzureResourceApiController.java

```diff
@@ -459,11 +479,47 @@ public ResponseEntity<ApiAzureIpResource> getAzureIp(UUID workspaceUuid, UUID re
     return new ResponseEntity<>(resource.toApiResource(), HttpStatus.OK);
   }
 
-  @Override
+  @Override // TODO get from LZ if it exists
   public ResponseEntity<ApiAzureRelayNamespaceResource> getAzureRelayNamespace(
       UUID workspaceId, UUID resourceId) {
     final AuthenticatedUserRequest userRequest = getAuthenticatedInfo();
     features.azureEnabledCheck();
+
+    // TODO take resource-id into consideration resourceId if there are multiple azure Relay
+    // instances
+    //  in the landing zone (no plan for this currently)
+    final AzureCloudContext azureCloudContext =
+        azureCloudContextService.getRequiredAzureCloudContext(workspaceId);
+    final bio.terra.landingzone.model.AzureCloudContext lzAzureCloudContext =
```

Comment:
Just for my own understanding -- are we inferring the landing zone from the tenant/subscription/mrg? This implies the mrg:landingZone relationship is 1:1.

Alternatively we could store the `landingZoneId` as part of the WSM `AzureCloudContext`. That would make things a little more explicit. Are there plans to do that or are we sticking with the 1:1 relationship for now?


are we storing the landingZoneId anywhere in the WSM workspace, or are we inferring the landing zone from the MRG?

Reply:
We are, I don't love it either. Right now we don't have landingZoneId stored in WSM, so this is future work we'll need.